.github/workflows/prerelease-tests.yml

```diff
@@ -0,0 +1,164 @@
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: pip
+          cache-dependency-path: |
+            mitosheet/setup.py
+            tests/requirements.txt
+            tests/extra-requirements.txt
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'npm'
+          cache-dependency-path: mitosheet/package-lock.json
+      - name: Install dependencies (ubuntu, macos)
+        if: matrix.os != 'windows-latest'
+        run: |
+          cd tests
+          python3 -m venv venv
+          source venv/bin/activate
+          pip install mitosheet streamlit
+          pip install --upgrade -i https://test.pypi.org/simple/ mitosheet
+          # Install necessary node packages
+          npm install
+          npx playwright install chromium webkit firefox chrome
+      - name: Setup streamlit (ubuntu, macos)
+        if: matrix.os != 'windows-latest'
+        run: |
+          mkdir -p ~/.streamlit/
+          echo "[general]"  > ~/.streamlit/credentials.toml
+          echo "email = \"\""  >> ~/.streamlit/credentials.toml
+      - name: Setup streamlit (windows)
+        if: matrix.os == 'windows-latest'
+        run: |
+          $streamlitDir = "$HOME\.streamlit"
+          if (-not (Test-Path -Path $streamlitDir)) {
+            New-Item -ItemType Directory -Path $streamlitDir
+          }
+          Set-Content -Path "$streamlitDir\credentials.toml" -Value @"
+          [general]
+          email = ""
+          "@
+      - name: Install dependencies (windows)
+        if: matrix.os == 'windows-latest'
+        run: |
+          cd tests
+          python3 -m venv venv
+          .\venv\Scripts\Activate.ps1
+          pip install mitosheet streamlit
+          pip install --upgrade -i https://test.pypi.org/simple/ mitosheet
+          # Install necessary node packages
+          npm install
+          npx playwright install chromium webkit firefox chrome
+      - name: Start a server and run tests (ubuntu, macos)
+        if: matrix.os != 'windows-latest'
+        run: |
+          cd tests
+          source venv/bin/activate
+          git clone https://github.com/mito-ds/every-feature-everywhere-all-at-once.git
+          cd every-feature-everywhere-all-at-once
+          streamlit run automation-app.py --server.port 8555 &
+          bash ../check_server.sh
+          npm run test:streamlit:demo -- --project="${{ matrix.project }}"
+      - name: Start a server and run tests (windows)
+        if: matrix.os == 'windows-latest'
+        run: |
+          cd tests
+          .\venv\Scripts\Activate.ps1
+
+          git clone https://github.com/mito-ds/every-feature-everywhere-all-at-once.git
+          cd every-feature-everywhere-all-at-once
+          Start-Job { streamlit run automation-app.py --server.port 8555 } -WorkingDirectory (Get-Location)
```

Comment:
Do we need to run a test on windows that the server is running?

Reply:
I think I just grabbed this from whatever we're currently doing. I guess the tests will catch if the server doesn't start up? Not totally suer why we need that in general